PhpAmqpLib/Connection/AbstractConnection.php

```diff
@@ -316,6 +316,38 @@ public function reconnect()
         $this->connect();
     }
 
+    /**
+     * Sets new password for this connection.
+     * Should be used for RabbitMQ's OAuth2 authentication backend to update current token.
+     * @param string $password
+     */
+    public function updatePassword($password): void
+    {
+        if ($this->login_method !== 'EXTERNAL') {
+            // password is always latest in response for PLAIN and AMQPPLAIN
+            // for EXTERNAL mechanism this has to be handled by calling user
+            $login_response_parts = explode("\0", $this->login_response);
+            $login_response_parts[count($login_response_parts) - 1] = $password;
+            $this->login_response = implode("\0", $login_response_parts);
+        }
+
+        if ($this->is_connected) {
+            // send new secret to broker if currently connected
+            $this->x_update_secret($password);
+        }
+
+        // for potential cloning
+        $this->replace_password_in_construct_params($password);
```

Comment:
Constructor params are deprecated and will be dropped in version 4.0

Reply:
When will version 4.0 released? Will version 4.0 handle OAuth?